### PR TITLE
feat: add envFrom value to Helm chart

### PR DESCRIPTION
**Pull request by ericpaulsen**
this PR adds support for the `coder.envFrom` value to the Helm chart. this will allow for Coder's environment variables to be passed from a single secret.

i'm pushing this in draft mode, because there is one caveat; the `$hasAccessURL` variable cannot identify if `CODER_ACCESS_URL` is set in the secret, so it will always default to the cluster-internal URL unless the access URL is specified using the traditional `coder.env` variable.

ideally, we parse the vars from the secret and pass them into the existing `containers.env` section in the template, but I have not gotten this to work. something akin to:

```tpl
{{ if .Values.coder.envFrom }}
{{- range $secretKey, $secretValue := .Values.coder.envFrom }}
# Handle the coder.envFrom case
{{- range $key, $value := dict $secretValue }}
- name: {{ $key.name }}
  value: {{ $value | b64dec | quote }}
{{- end }}
{{- end }}
{{- end }}
```

### CODE REVIEW

File: helm/coder/values.yaml [hunk 1]
```diff
@@ -21,6 +21,10 @@ coder:
   # - name: "CODER_ACCESS_URL"
   #   value: "https://coder.example.com"
 
+  envFrom:
```
Comment (deansheather) [inline helm/coder/values.yaml:24]: needs a comment in the same format as all other values

File: helm/coder/values.yaml [hunk 2]
```diff
@@ -21,6 +21,10 @@ coder:
...
+  - secretRef:
+      name: ""
```
Comment (deansheather) [inline helm/coder/values.yaml:26]: these need to be commented out before merge

File: helm/coder/templates/_coder.tpl [hunk 3]
```diff
@@ -30,6 +30,10 @@ args:
   {{- end }}
 - server
 {{- end }}
+envFrom:
+{{- with .Values.coder.envFrom }}
+{{ toYaml . }}
```
Comment (deansheather) [inline helm/coder/templates/_coder.tpl:35]: Do you need an `nindent 2` here?

**Review by deansheather — COMMENTED** [review]
w.r.t the CODER_ACCESS_URL being automatically set, you could just put that behind a new value too like `coder.useClusterAccessURL` or something

File: helm/coder/values.yaml [hunk 4]
```diff
@@ -29,13 +40,13 @@ coder:
     # if not set. If you're using the chart directly from git, the default
     # app version will not work and you'll need to set this value. The helm
     # chart helpfully fails quickly in this case.
-    tag: ""
+    tag: "v2.1.5"
```
Comment (spikecurtis) [inline helm/coder/values.yaml:43]: This isn't supposed to be set in the repo

File: helm/coder/values.yaml [hunk 5]
```diff
@@ -29,13 +40,13 @@ coder:
...
     # coder.image.pullPolicy -- The pull policy to use for the image. See:
     # https://kubernetes.io/docs/concepts/containers/images/#image-pull-policy
     pullPolicy: IfNotPresent
     # coder.image.pullSecrets -- The secrets used for pulling the Coder image from
     # a private registry.
-    pullSecrets: []
+    pullSecrets:
```
Comment (spikecurtis) [inline helm/coder/values.yaml:49]: why is this changed?

**Review by spikecurtis — DISMISSED** [review]
Please add a test that uses this with a new golden file

**Review by deansheather — COMMENTED** [review]
LGTM but I contributed so I won't approve

**Review by spikecurtis — APPROVED** [review]

**Pull request merged**